lib/state-service.js

```diff
@@ -0,0 +1,122 @@
+        .items(featureShema),
+    strategies: joi
+        .array()
+        .optional()
+        .items(strategySchema),
+});
+
+class StateService {
+    constructor(config) {
+        this.config = config;
+    }
+
+    async importFile({ importFile, dropBeforeImport, userName }) {
+        let data = await new Promise((resolve, reject) =>
+            fs.readFile(importFile, (err, v) =>
+                err ? reject(err) : resolve(v)
+            )
+        );
+        if (mime.lookup(importFile) === 'text/yaml') {
+            data = yaml.safeLoad(data);
+        }
+
+        await this.import({
+            data,
+            dropBeforeImport,
+            userName,
+        });
+    }
+
+    async import({ data, userName, dropBeforeImport }) {
+        const { eventStore } = this.config.stores;
+
+        if (typeof data === 'string') {
+            data = JSON.parse(data);
+        }
+
+        data = await joi.validate(data, dataSchema);
+
+        if (data.features) {
+            logger.info(`Importing ${data.features.length} features`);
+            if (dropBeforeImport) {
+                logger.info(`Dropping existing features`);
+                await eventStore.store({
+                    type: DROP_FEATURES,
+                    createdBy: userName,
+                    data: { name: 'all-features' },
+                });
+            }
+            for (const feature of data.features) {
+                await eventStore.store({
+                    type: FEATURE_IMPORT,
+                    createdBy: userName,
+                    data: feature,
+                });
+            }
+        }
+
+        if (data.strategies) {
+            logger.info(`Importing ${data.strategies.length} strategies`);
+            if (dropBeforeImport) {
+                logger.info(`Dropping existing strategies`);
+                await eventStore.store({
+                    type: DROP_STRATEGIES,
+                    createdBy: userName,
+                    data: { name: 'all-strategies' },
+                });
+            }
+            for (const strategy of data.strategies) {
+                await eventStore.store({
+                    type: STRATEGY_IMPORT,
+                    createdBy: userName,
+                    data: strategy,
+                });
+            }
+        }
+    }
+
+    async export({ strategies, featureToggles }) {
+        const { featureToggleStore, strategyStore } = this.config.stores;
+        const result = {};
+
+        if (featureToggles) {
+            result.features = await featureToggleStore.getFeatures();
+        }
+
+        if (strategies) {
+            result.strategies = (await strategyStore.getStrategies())
+                .filter(strat => strat.editable)
+                .map(strat => {
+                    strat = Object.assign({}, strat);
+                    delete strat.editable;
+                    return strat;
+                });
+        }
+
+        return result;
```

Comment:
You can use Promise.all to fetch both strategies and toggles in paralell

```js
const [features, strategies ] = await Promise.all([
    features ? featureToggleStore.getFeatures() : Promise.resolve(),
    strategies ? strategyStore.getStrategies({builtIn: false}) : Promise.resolve(),
 ]);

return {strategies, features};
```

Reply:
and we could even return the promise instead of awaiting it:

```
return Promise.all([
    features ? featureToggleStore.getFeatures() : Promise.resolve(),
    strategies ? strategyStore.getStrategies({builtIn: false}) : Promise.resolve(),
 ]).then([features, strategies ] => {features, strategies});
```